pkg/rbac/namespacedAuthorization.go

```diff
@@ -0,0 +1,86 @@
+package rbac
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	"k8s.io/klog/v2"
+)
+
+type namespacedResources struct {
+	err       error
+	resources map[string]map[string]map[string]string //["namespace":["group1": ["kind", "kind2",..]], "namespace1": ["group1" : ["kind1", "kind2"]]]
+	updatedAt time.Time
+	lock      sync.Mutex
+}
+
+func (cache *Cache) NamespacedResources(ctx context.Context) (map[string]map[string]map[string]string, error) {
+	namespaced, err := cache.users.getNamespacedResources(cache, ctx)
+	return namespaced, err
+
+}
+
+func (users *namespacedResources) getNamespacedResources(cache *Cache, ctx context.Context) (map[string]map[string]map[string]string, error) {
+
+	users.lock.Lock()
+	defer users.lock.Unlock()
+	if users.resources != nil &&
+		time.Now().Before(users.updatedAt.Add(time.Duration(config.Cfg.UserCacheTTL)*time.Millisecond)) {
+		klog.V(5).Info("Using user's namespaced resources from cache.")
+		return users.resources, users.err
+
+	}
+
+	resourceMap := make(map[string]map[string]map[string]string)
+	klog.V(5).Info("Getting namespaced resources from ")
+	resources, err := listResources()
+	if err != nil {
+		klog.Errorf("Error getting resources", err)
+	}
+	for _, resources := range resources {
+		resourceMap["namespace"] = make(map[string]map[string]string)
+		resourceMap["namespace"]["apigroupd"] = make(map[string]string)
+		resourceMap["namespace"]["apigroupd"]["kinds"] = resources.Kind
+
+	}
+	fmt.Println("Resources:", resourceMap)
+
+	return users.resources, users.err
+}
+
+func listResources() ([]metav1.APIResource, error) {
```

Comment:
For Namespaced resources we don't need the resources ahead.  This is different than the non-namespaced.

We only need the projects(namespaces) first, then we do the equivalent of
```
# For each namespace:
oc auth can-i --list -n <project-name> --as=user
```

For cluster-scoped we need to know all the resources ahead because we need to do the equivalent of
```
# For each cluster-scoped resource
oc auth can-i list <apigroup.kind> --as=user
```